hikari/impl/rest.py

```diff
@@ -3712,6 +3712,19 @@ async def fetch_application_command(
             response, guild_id=snowflakes.Snowflake(guild) if guild is not undefined.UNDEFINED else None
         )
 
+    def _deserialize_command_list(
+        self, command_payloads: data_binding.JSONArray, guild_id: typing.Optional[snowflakes.Snowflake]
+    ) -> typing.List[commands.PartialCommand]:
+        command_objs: typing.List[commands.PartialCommand] = []
+        for payload in command_payloads:
+            try:
+                command_objs.append(self._entity_factory.deserialize_command(payload, guild_id=guild_id))
+
+            except errors.UnrecognisedEntityError:
+                pass
+
+        return command_objs
```

Comment:
Why not allow extra values to pass to the function in `data_binding.cast_enum_array` so that we can get rid of this?

Reply:
taking `*args, **kwargs` for everything which uses it seems a bit inefficient when it isn't meant to meet every use case and practically never could (this is just 1 very specific use case); it doesn't cover the (more common) case where arrays of raw variants are deserialized to `dict[id, Unique]` partially cause out of 4 cases only 3 of them would be covered by a helper function as the 4th uses a type assertion which'd be left out due to avoiding unnecessary complexity (3 uses being less than half of the amount of times `cast_variants_array` is used)